x-pack/test/api_integration/deployment_agnostic/apis/observability/ai_assistant/complete/functions/context.spec.ts

```diff
@@ -0,0 +1,240 @@
+    text: 'My favourite color is blue.',
+  },
+  {
+    id: 'alert_instructions',
+    title: 'Alert Handling Guide',
+    text: 'All alerts should be considered high priority. Every alert is monitored every day. Threshold alerts should be resolved first. Consider this when analyzing alerts.',
+  },
+  {
+    id: 'miscellaneous',
+    title: 'Miscellaneous Note',
+    text: 'hello again',
+  },
+];
+
+const userPrompt = `What's my favourite color?`;
+
+export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const observabilityAIAssistantAPIClient = getService('observabilityAIAssistantApi');
+  const es = getService('es');
+  const ml = getService('ml');
+  const log = getService('log');
+
+  describe('context', function () {
+    this.tags(['failsOnMKI']);
+    let llmProxy: LlmProxy;
+    let connectorId: string;
+    let messageAddedEvents: MessageAddEvent[];
+
+    before(async () => {
+      llmProxy = await createLlmProxy(log);
+      connectorId = await observabilityAIAssistantAPIClient.createProxyActionConnector({
+        port: llmProxy.getPort(),
+      });
+
+      await addSampleDocsToInternalKb(getService, sampleDocsForInternalKb);
+
+      void llmProxy.interceptWithFunctionRequest({
+        name: CONTEXT_FUNCTION_NAME,
+        arguments: () => JSON.stringify({}),
+        when: () => true,
+      });
+
+      void llmProxy.interceptConversation('Your favourite color is blue.');
+
+      ({ messageAddedEvents } = await chatComplete({
+        userPrompt,
+        screenContexts,
+        connectorId,
+        observabilityAIAssistantAPIClient,
+      }));
+
+      await llmProxy.waitForAllInterceptorsToHaveBeenCalled();
+    });
+
+    after(async () => {
+      llmProxy.close();
+      await observabilityAIAssistantAPIClient.deleteActionConnector({
+        actionId: connectorId,
+      });
+
+      await deleteKnowledgeBaseModel(ml);
+      await deleteInferenceEndpoint({ es });
+      await clearKnowledgeBase(es);
+    });
+
+    describe('calling the context function via /chat/complete', () => {
+      let firstRequestBody: ChatCompletionStreamParams;
+      let secondRequestBody: ChatCompletionStreamParams;
+
+      before(async () => {
+        firstRequestBody = llmProxy.interceptedRequests[0].requestBody;
+        secondRequestBody = llmProxy.interceptedRequests[1].requestBody;
+      });
+
+      it('makes 2 requests to the LLM', () => {
+        expect(llmProxy.interceptedRequests.length).to.be(2);
+      });
+
+      it('emits 3 messageAdded events', () => {
+        expect(messageAddedEvents.length).to.be(3);
+      });
+
+      describe('The first request - Scoring documents', () => {
+        it('contains the correct number of messages', () => {
+          expect(firstRequestBody.messages.length).to.be(2);
+        });
+
+        it('contains the system message as the first message in the request', () => {
+          expect(firstRequestBody.messages[0].role === MessageRole.System);
```

Comment:
nit: Let's try to use `first(...)` and `last(...)` helpers to make it clear we are not picking arbitrary messages.

```suggestion
          expect(firstRequestBody.messages[0].role === MessageRole.System);
```